uqcsbot/base.py

```diff
@@ -39,10 +50,12 @@ def _handle_command(self, event_data: dict) -> None:
         channel = Channel(self.client, message["channel"])
         command = Command(command_name, None if not arg else arg[0], channel)
         for cmd in self._command_registry[command_name]:
-            cmd(command)
+            asyncio.ensure_future(cmd(command), self.evt_loop)
 
     def on_command(self, command_name: str):
         def decorator(fn):
+            if not asyncio.iscoroutinefunction(fn):
```

Comment:
Just to confirm that I understand this correctly, is this code here to allow you to optionally make async scripts?

Reply:
It allows you to _write_ it optionally async, everything will be run async though